prog/vnet/subnet_nexus.rb

```diff
@@ -38,13 +38,90 @@ def wait
       hop :refresh_mesh
     end
 
+    when_refresh_keys_set? do
+      ps.update(state: "refreshing_keys")
+      hop :refresh_keys
+    end
+
     nap 30
   end
 
+  def gen_encryption_key
+    "0x" + SecureRandom.bytes(36).unpack1("H*")
+  end
+
+  def gen_spi
+    "0x" + SecureRandom.bytes(4).unpack1("H*")
+  end
+
+  def gen_reqid
+    SecureRandom.random_number(100000) + 1
+  end
+
+  def refresh_keys
+    payload = {}
+
+    ps.nics.each do |nic|
+      payload[nic.id] = {
+        spi4: gen_spi,
+        spi6: gen_spi,
+        reqid: gen_reqid
+      }
+      nic.update(encryption_key: gen_encryption_key)
+    end
+
+    ps.nics.each do |nic|
+      bud Prog::Vnet::RekeyNic, {payload: payload, subject_id: nic.id}
+    end
+
+    hop :wait_inbound_setup
+  end
+
+  def wait_inbound_setup
+    reap
+    DB.transaction do
+      if strand.children.filter { |stc| stc.prog == "Vnet::RekeyNic" }.all? { |stc| stc.label == "wait_outbound_trigger" }
+        ps.nics.each do |nic|
+          nic.incr_trigger_outbound_update
+        end
+        hop :wait_outbound_setup
+      end
+    end
+
+    donate
+  end
+
+  def wait_outbound_setup
+    reap
+    DB.transaction do
+      if strand.children.filter { |stc| stc.prog == "Vnet::RekeyNic" }.all? { |stc| stc.label == "wait_state_drop_trigger" }
```

Comment:
Wew, having to do this sure looks like a weakness in the programming model. Either we're missing something there, or this code is very special, or we should do something different (exact timeline undefined, it's acceptable to put this scar in and think about how to reverse it later)

Reply:
new PR doesn't have this https://github.com/ubicloud/ubicloud/pull/441